auth/oras/client.go

```diff
@@ -22,7 +24,9 @@ func NewClient(opts ClientOptions) remote.Client {
 	client := &auth.Client{
 		Cache:    auth.NewCache(),
 		ClientID: "acr-cli",
+		Header:   http.Header{},
```

Comment:
You may consider start to add a unit test for client and validate the change.

Reply:
We tested by running under debug mode and looked into the http requests and server logs. Do we still need a unit test?

Reply:
> We tested by running under debug mode and looked into the http requests and server logs.

Those are manual testing.

> Do we still need a unit test?

Yes, we need an auto way to test. It's not only for the completeness but also to prevent future regressions.

Reply:
Added one unit test.